### ISSUE TITLE
Adding ability to specify pronouns for a user profile

### ISSUE THREAD

**Issue by KCony** [Submitty/Submitty#8735]
**What problem are you trying to solve with Submitty**
Currently, Submitty does not allow users to specify their pronouns in the profile. See the screenshot.

**Describe the way you'd like to solve this problem**
In order to promote diversity and inclusion, I strongly believe that adding the ability for a user to specify their pronouns is a must. This way, teaching staff will be able to look up this information in the student's profile and know how to refer to them correctly.

**Describe any potential alternatives you'd tried to solve the problem**
The only alternative would be for the teaching staff to reach out to students directly asking for their pronouns and then storing this information locally for a given class and outside of Submitty. This is inconvenient, insecure, and, generally, awkward.

**Additional context**
See the screenshot below:
<img width="1160" alt="Screenshot 2023-01-12 114029" src="https://user-images.githubusercontent.com/7108039/212127698-f2e3057d-0a9d-4220-a783-ca27e0c072af.png">

**Comment by bmcutler** [comment]
Putting a few links here on UI design (there are many more resources for creating a good design).

* https://www.gsrc.princeton.edu/creating-inclusive-forms#:~:text=You%20can%20ask%20pronouns%20by,someone%20in%20the%20third%20person

* https://uxdesign.cc/designing-forms-for-gender-diversity-and-inclusion-d8194cf1f51

Most important:  this data is optional for each user to specify.  This column should default to empty string.

The information should be displayed as a column on the "Manage Students" page that only instructor user can access.  

The Manage Students page has many columns (and even more information is available about each user when you click "Edit" for a particular user (e.g., "User Numeric ID" and "Secondary Email").   We are running out of reasonable horizontal space on this page.

So I would like to ask for an Additional Related Feature Request: Let's allow the instructor to toggle which of these many columns of data to display on the Manage Students page.

**Comment by mahi-pas** [comment]
Can I be assigned to this issue? @bmcutler

**Comment by moonLight-7k** [comment]
Hello, I am interested in contributing to open source projects and would love to participate in any that you have available. I have experience in web Development and am eager to learn and grow my skills through working on these projects. Please let me know if there are any opportunities for me to get involved, I would greatly appreciate it. Thank you!

**Comment by JaySingh23** [comment]
Hi @KCony, I am new at open source development. This would be my first contribution in any open source project and I would be glad if I am assigned to help fix this issue.

I read the articles as mentioned by the user @bmcutler. I am thinking of creating radio buttons which will have "MALE", "FEMALE" and "Prefer not to say" as options. This will be a quick to implement approach and would do the work. Or maybe instead of the male, female we can give them the option of choosing pronouns like "he/him" and "she/her". 

Please tell me what would be the best option and if possible then please do assign me this. I would really appreciate it. Thanks.

**Comment by mahi-pas** [comment]
We should follow this design standard, as it is the most inclusive: https://designsystem.digital.gov/patterns/create-a-user-profile/pronouns/

TLDR: Make an input box where users can type in their own pronouns. There is also a checkbox that says "Prefer not to share my pronouns." This checkbox can be omitted, as a blank field is functionally the same and simpler to implement.

**Comment by shailpatels** [comment]
Hi @JaySingh23 @mahi-pas, due to how our GitHub organization is set up we cannot assign external users to an issue. But if no one else is assigned/working on it feel free to take it! If you make a pull request we would be happy to go through the review process and try and get it merged.

Please read the suggestions for new developers here: https://submitty.org/developer/getting_started/index and try and get a developer instance of Submitty running locally.

We cannot assign an issue to someone outside of the Submitty GitHub project, but if you make a PR we would be happy to review it. Please try and look at the documentation at submitty.org and if you get stuck on something ask on the general chat in [slack here](https://join.slack.com/t/submitty/shared_invite/enQtMzE1NzgyMzUzNzI5LWZhOGM4NWU4M2U3ZmM2ZmM5NTYwNjk3NTc5Yjg1ZGRlYzU1MjcwYWNkNGYxZjk2YTk3ZmEyODY3MjEyOTc0M2E).

> We should follow this design standard, as it is the most inclusive: https://designsystem.digital.gov/patterns/create-a-user-profile/pronouns/
> 
> TLDR: Make an input box where users can type in their own pronouns. There is also a checkbox that says "Prefer not to share my pronouns." This checkbox can be omitted, as a blank field is functionally the same and simpler to implement.

Agreed, this would be the preferred way to take input for something like pronouns.

Also I would suggest adding a notice to the user on who will be able to see the preferred pronouns if set (only instructors? all graders?, etc)

**Comment by mahi-pas** [comment]
> Also I would suggest adding a notice to the user on who will be able to see the preferred pronouns if set (only instructors? all graders?, etc)

That's a great idea! What are the access levels that we can set? Is there a precedent for it or will we have to write the privacy implementation on our own?

**Comment by bmcutler** [comment]
In my comment above I indicated it would only be available to instructors.  But we can revising this if a compelling need for using it elsewhere on the site was identified.

**Comment by depleur** [comment]
> In my comment above I indicated it would only be available to instructors. But we can revisit this if a compelling need for using it elsewhere on the site was identified.

Why not make it up to the student to set the visibility of their pronouns? A drop-down menu to implement this alongside the menu to select pronouns would be a good addition, in my opinion. 

Maybe having a couple of options like:

- Set to instructors only
- Set to instructors and peers

could work?

**Comment by mohamedsaajid** [comment]
Hey @KCony and @mahi-pas , I would like to contribute to this issuse, in my opinion a dropdown menu would be the most appropriate and the most modern menu to specify pronouns.

**Comment by bmcutler** [comment]
this was closed in PR #8921 (but unfortunately it wasn't linked & closed automatically)
